astropy/stats/bls/core.py

```diff
@@ -758,6 +757,8 @@ def assert_allclose(self, other, **kwargs):
             The other results object to compare.
 
         """
+        from ...tests.helper import assert_quantity_allclose
```

Comment:
Is the ``assert_allclose`` actually meant to be used by users or is it just for the tests? (maybe @dfm can clarify?)

If it's meant to be used by users, I wonder if we should instead import ``from astropy.units.quantity import _unquantify_allclose_arguments`` and use ``np.assert_allclose`` here? (to avoid the pytest dependency)

Reply:
Usually, "assert" is used in testing. In a non-testing case, maybe use `is_allclose` that returns a boolean instead, or maybe that ship has sailed now...?

Reply:
it hasn't sailed at all, the bls module is only in dev so any fixes should be done _now_

Reply:
actually, @dfm, I wonder whether we need this convenience at all?

Reply:
Hi team! This *is* just for testing and it's not meant as a user facing method... perhaps we should move it to the tests script anyways?

Reply:
@dfm - thanks for confirming - indeed this should definitely be moved inside the ``tests`` directories.